perun-core/src/main/java/cz/metacentrum/perun/core/api/ResourcesManager.java

```diff
@@ -65,6 +65,20 @@
 	 */
 	Resource createResource(PerunSession perunSession, Resource resource, Vo vo, Facility facility) throws InternalErrorException, PrivilegeException, VoNotExistsException, FacilityNotExistsException, ResourceExistsException;
 
+	/**
+	 * Copy "template" settings(attributes,services,..) from user's another existing resource and create new resource with this template.
+	 * Resource can be from any of user's facilities.
+	 *
+	 * @param perunSession
+	 * @param resource template resource to copy
+	 * @param newResourcename name of the new Resource
+	 * @param destinationVo virtual organization where the new resource will be created
+	 * @param destinationFacility facility where the new resource will be created.
+	 * @throws InternalErrorException
+	 * @throws ResourceNotExistsException
+	 */
+	Resource copyResource(PerunSession perunSession, Resource resource, String newResourcename, Vo destinationVo, Facility destinationFacility) throws InternalErrorException, ResourceNotExistsException, FacilityNotExistsException, PrivilegeException, VoNotExistsException, ResourceExistsException;
```

Comment:
I think you should use "destinationResource" instead of VO + resourceName + destinationFacility because destinationResource object will have all these information. I am also missing the parameter "withGroups" if both resources are from the same Vo.